src/main/java/com/ludo/study/studymatchingplatform/study/domain/study/Study.java

```diff
@@ -45,294 +30,339 @@
+    }
+
+    public void removeParticipant(final Participant participant) {
+        participants.removeIf(p -> Objects.equals(p, participant));
+    }
+
+    public void modifyStatus(final StudyStatus status, final LocalDateTime now) {
+        // 모집 마감 상태는 시간에 따라 다른 결과 반영
+        if (status == StudyStatus.RECRUITED) {
+            modifyStatusToRecruited(now);
+            modifyStatusToProgress(now);
+        }
+    }
+
+    public void modifyStatusToRecruiting() {
+        this.status = StudyStatus.RECRUITING;
+    }
+
+    private void modifyStatusToRecruited(final LocalDateTime now) {
+        if (this.startDateTime.isAfter(now)) {
+            this.status = StudyStatus.RECRUITED;
+        }
+    }
+
+    private void modifyStatusToProgress(final LocalDateTime now) {
+        if (this.startDateTime.isBefore(now) && this.endDateTime.isAfter(now)) {
+            this.status = StudyStatus.PROGRESS;
+        }
+    }
+
+    public void modifyStatusToCompleted(final LocalDateTime now) {
+        if (this.endDateTime.isBefore(now)) {
+            this.status = StudyStatus.COMPLETED;
+        }
+    }
+
+    public void deactivateForRecruitment() {
+        this.recruitment.softDelete();
+    }
+
+    public void activateForRecruitment() {
+        this.recruitment.activate();
+    }
+
+    public Boolean ensureHasRecruitment() {
+        if (this.recruitment != null && !this.recruitment.isDeleted()) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
+    public void update(final String title, final Category category, final Integer participantLimit,
+                       final Way way, final Platform platform, final String platformUrl,
+                       final LocalDateTime startDateTime, final LocalDateTime endDateTime) {
+        if (title != null) {
+            this.title = title;
+        }
+        if (category != null) {
+            this.category = category;
+        }
+        if (participantLimit != null) {
+            this.participantLimit = participantLimit;
+        }
+        if (way != null) {
+            this.way = way;
+        }
+        if (platform != null) {
+            this.platform = platform;
+        }
+        if (platformUrl != null) {
+            this.platformUrl = platformUrl;
+        }
+        if (startDateTime != null) {
+            this.startDateTime = startDateTime;
+        }
+        if (endDateTime != null) {
+            this.endDateTime = endDateTime;
+        }
+    }
+
+    public void ensureReviewPeriodAvailable(final LocalDateTimePicker localDateTimePicker) {
+        final LocalDateTime now = localDateTimePicker.now();
+        final LocalDateTime reviewAvailStartTime = endDateTime.plusDays(3).minusSeconds(1);
+        final LocalDateTime reviewAvailEndTime = endDateTime.plusDays(14).plusSeconds(1);
+
+        // Order Specific
+        if (isReviewPeriodElapsed(now)) {
```

Comment:
```suggestion
        if (isAvailableReviewPeriod()) {
            return;
        }
        throw new InvalidReviewPeriodException("리뷰 작성 기간이 아닙니다.", reviewAvailStartTime, reviewAvailEndTime);
```

리뷰 가능 시간에 대한 조건을 and 조건으로 묶으면 `Order Specific` 에서 조금 자유로워질 것 같습니다.
추 후 `리뷰 가능 시간 요구사항이` 변경되어도, `isAvailableReviewPeriod()` 메서드 하나만 수정하면 되기 때문에 유지보수 측면에서 장점이 있을 것으로 생각됩니다 🙂

Reply:
commit에 suggestion 반영하였습니다.👍